### PR TITLE
Support mysql BINARY, VARBINARY and Postgres Type::BYTEA

### PR DESCRIPTION
**Pull request by Sevenannn**
Support mysql BINARY, VARBINARY and Postgres Type::BYTEA for the mysql table provider and postgres table provider
* mysql `BINARY` & `CHAR` are both of type `MYSQL_TYPE_STRING` in mysql-async library. Add additional binary flag data to determine mysql binary / char and cast them to arrow binary / largestring accordingly

### CODE REVIEW

File: src/sql/arrow_sql_gen/mysql.rs [hunk 1]
```diff
@@ -275,6 +273,10 @@ pub fn rows_to_arrow(rows: &[Row]) -> Result<RecordBatch> {
                         i
                     );
                 }
+                column_type @ (ColumnType::MYSQL_TYPE_STRING
```
Comment (sgrebnov) [inline src/sql/arrow_sql_gen/mysql.rs:276]: 1. Is this for acceleration? 
2. Could you please verify that query returns the same result as direct query
3. As we are changing to MYSQL_TYPE_STRING to be binary I propose to change  MYSQL_TYPE_STRING->MYSQL_TYPE_VARCHAR for `char`
_ if data_type.starts_with("char") => ColumnType::MYSQL_TYPE_STRING,

Comment (sgrebnov) [reply src/sql/arrow_sql_gen/mysql.rs:276]: Example query
image

**Review by sgrebnov — APPROVED** [review]

File: src/sql/arrow_sql_gen/mysql.rs (hunk 1 above)
Comment (Sevenannn) [reply src/sql/arrow_sql_gen/mysql.rs:276]: This is for acceleration, the verified data can be seen from spice oss ticket

Comment (Sevenannn) [reply src/sql/arrow_sql_gen/mysql.rs:276]: Checked with the new fix, it doesn't broke query in spice repl, see the query result in: https://github.com/spiceai/spiceai/issues/1871

Comment (Sevenannn) [reply src/sql/arrow_sql_gen/mysql.rs:279]: Added new code to detect whether the column is binary (BINARY, VARBINARY) or non binary (CHAR VARCHAR) . The change don't have any degrading behavior tested on tpch queries

**Review by sgrebnov — APPROVED** [review]
@Sevenannn  -  great catch on BINARY_FLAG 👍

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**